src/Kestrel.Core/Internal/Http/Http1Connection.FeatureCollection.cs

```diff
@@ -39,15 +38,6 @@ public partial class Http1Connection : IHttpUpgradeFeature
 
             StatusCode = StatusCodes.Status101SwitchingProtocols;
             ReasonPhrase = "Switching Protocols";
-            ResponseHeaders["Connection"] = "Upgrade";
```

Comment:
I'm not sure we're supposed to be removing the Connection: Upgrade response header.

/cc @Tratcher @halter73

Reply:
Websockets set that, this is redundant

Reply:
Who's job is it to set it in general? Why shouldn't the server set the response header? I get removing the other piece but this seems like it should be the case if an upgrade was requested, not a connection close header.

Reply:
Websocket is the only consumer so there's no clear boundary. Websocket sets a list of headers it needs, including this one.

Reply:
This is super pedantic but I would keep the connection header, it’s more correct imo. Getting a connection close header after calling upgrade is just incorrect.

Reply:
Agree, updated commits - haven't removed response header of connection upgrade. No change in test cases required.